tests/instruments/test_instrument.py

```diff
@@ -750,3 +765,40 @@ def test_attribute_access(self, inst):
 
     def test_len(self, inst):
         assert len(inst.channels) == 2
+
+
+class TestMultiFunctionality:
+    """Test the usage of children for different functionalities."""
+    class SomeFunctionality(Channel):
+        """This Functionality needs a prepended `id`."""
+
+        def write(self, command, **kwargs):
+            super().write(f"{self.id}:{command}")
+
+        voltage = Channel.control("Volt?", "Volt %f", "Set voltage in Volts.")
+
+    class InstrumentWithFunctionality(ChannelInstrument):
+        """Instrument with some functionality."""
+
+        def __init__(self, adapter, **kwargs):
+            super().__init__(adapter, **kwargs)
+            self.add_child(TestMultiFunctionality.SomeFunctionality, "X", "functions", "f")
```

Comment:
Minor: you could use kwargs here to make the role of the different args clearer.